parallel-install/pkg/components/component.go

```diff
@@ -42,10 +43,11 @@ type KymaComponent struct {
 //"chartDir" is a local filesystem directory with the component's chart.
 //
 //"overrides" is a function that returns overrides for the release.
-func NewComponent(name, namespace, chartDir string, overrides func() map[string]interface{}, helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
+func NewComponent(name, namespace, profile, chartDir string, overrides func() map[string]interface{}, helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
```

Comment:
This is a quite long parameter list.. wouldn't it be better to use a param-struct to make it easier to read?

Reply:
Example:

There are actually two options (at least I have in mind) to address long/complex parameter lists:

**Option 1. Param object**

        #Before
        func NewComponent(name, namespace, profile, chartDir string, overrides func() map[string]interface{},         helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
       ...
       }
        
        # With parameter object:
        struct NewComponentParams  struct {
           Name string
           Namespace string
           Profile string
           Chart string
           Overrides func() map[string]interface{}
           HelmClient helm.ClientInterface
           Log func(string, ...interface{}
        }
        func NewComponent(params NewComponentParams) {...}

**Option 2. Builder pattern**

        #Before
        func NewComponent(name, namespace, profile, chartDir string, overrides func() map[string]interface{}, helmClient helm.ClientInterface, log func(string, ...interface{})) *KymaComponent {
        ...
        }

        # a) Declarative way:
        compBuilder := ComponentBuilder{
          Name: "the name",
          Namespace: "theNamespace",
          EvaluationProfile: true,
          ChartName: "..."
          ...
        }
        comp := compBuilder.build()
        
       # b) Functional way
       compBuilder := ComponentBuilder{}
       comp := compBuilder.name("thename").
           namespace("thenamespace").
           useEvaluationProfile().
           chartNAme("chartanme").
           withHelmclient(helmclient).
           build()